sos/plugins/ipa.py

```diff
@@ -120,14 +120,24 @@ def setup(self):
         self.add_forbidden_path("/etc/dirsrv/slapd-*/pwdfile.txt")
         self.add_forbidden_path("/etc/named.keytab")
 
+        ### Make sure to lookup the right NSS DB for PKI certs
+        if ipa_version == "v3":
+            self.add_cmd_output([
+                "certutil -L -d %s/alias" % self.pki_tomcat_dir_v3
+            ])
+        elif ipa_version == "v4":
+            self.add_cmd_output([
+                "certutil -L -d %s/alias" % self.pki_tomcat_dir_v4
```

Comment:
Ditto.

Reply:
Considering that these two blocks are so similar though, why not just assign `pki_tomcat_dir` inside the conditionals, and then add the command once: this means there is only one copy of the (identical) command string to maintain and it prevents any sync problems when someone edits the file in future.

Reply:
You could also then use a conditional assignment and get rid of the blocks entirely:

```
        pki_tomcat_dir = self.pki_tomcat_dir_v4 if ipa_version == "v4" else self.pki_tomcat_dir_v3
        self.add_cmd_output("certutil -L -d %s/alias" % self.pki_tomcat_dir)
```

Reply:
In fact (although it's not code changed in this patch): there is no good reason for either of the `pki_tomcat_dir` variables to be instance members of the plugin class. They are constants. They can be set at file scope and then referenced throughout the plugin without any need for the `self.` prefix (which causes line lengths to expand).

Reply:
ack. makes sense. I changed the patch and also included a change for the configuration files. The variables pki_tomcat_dir and pki_tomcat_conf_dir are still instance members of the plugin class though. Maybe this can be changed with a later commit. I think the whole plugin needs some more work.